x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/LimitedRole.java

```diff
@@ -74,6 +76,22 @@ public RemoteIndicesPermission remoteIndices() {
         throw new UnsupportedOperationException("cannot retrieve remote indices permission on limited role");
     }
 
+    @Override
+    public boolean hasWorkflowsRestriction() {
+        return baseRole.hasWorkflowsRestriction() || limitedByRole.hasWorkflowsRestriction();
+    }
+
+    @Override
+    public Role forWorkflow(Workflow workflow) {
+        Role baseRestricted = baseRole.forWorkflow(workflow);
+        if (baseRestricted == EMPTY_RESTRICTED_BY_WORKFLOW) {
+            return EMPTY_RESTRICTED_BY_WORKFLOW;
+        }
+        // We don't support workflows restriction for limitedByRole,
+        // hence it is safe to return this object unchanged.
+        return this;
```

Comment:
I don't know why it is safe to return unchanged? Checking whether limitedByRole is empty and restricted may be redundant but feels technically "safer" to me. Also the `Role.limitedBy(..)` does not prevent it so I think we should be consistent.

Reply:
You are right. It's safer to add an additional check instead of to assume.